### PR TITLE
Move HBaseTableUtil to a separate module so that other modules that w…

### PR DESCRIPTION
**Pull request by gokulavasan**
…ants to use HBase Tables can do so without depending on data-fabric. This is in preparation of Transaction messaging service work which needs to use HBase tables but doesn't need or want to depend on data-fabric.

Build : http://builds.cask.co/browse/CDAP-DUT5050

### CODE REVIEW

File: cdap-hbase-compat-1.2-cdh5.7.0/pom.xml [hunk 1]
```diff
@@ -50,6 +50,13 @@
   </dependency>
   <dependency>
     <groupId>co.cask.cdap</groupId>
+    <artifactId>cdap-hbase-compat-base</artifactId>
```
Comment (chtyim) [inline cdap-hbase-compat-1.2-cdh5.7.0/pom.xml:58]: All the hbase-compat modules should have a dependency on `cdap-hbase-compat-base`, isn't it? It is currently coming from the transitive dependency through data-fabric, but it is better to always declare the direct dependency that this module is actually depending upon.

File: cdap-hbase-compat-base/pom.xml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      <groupId>org.apache.hbase</groupId>
+      <artifactId>hbase-server</artifactId>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+    </dependency>
+    <!-- Test dependencies -->
```
Comment (chtyim) [inline cdap-hbase-compat-base/pom.xml]: Junit is also test dependencies (the scope is already set in the parent pom, but this comment should move up)

File: cdap-hbase-compat-1.2-cdh5.7.0/pom.xml (hunk 1 above)
Comment (gokulavasan) [reply cdap-hbase-compat-1.2-cdh5.7.0/pom.xml:58]: I see ok. I will declare direct dependency on cdap-hbase-compat-base for all compat modules.

**Comment by chtyim** [comment]
LGTM.

**Pull request merged**